src/prog_models/prognostics_model.py

```diff
@@ -828,11 +827,19 @@ def update_all():
                 times.append(t)
                 inputs.append(u)
                 states.append(deepcopy(x))  # Avoid optimization where x is not copied
+
+        if callable(config['dt']):
```

Comment:
Is there a short comment that would be useful to add before this "if" statement? Something like, "configuring next_time function to define prediction time step, default is constant"